### ISSUE TITLE
Default cipher list to updated mozilla intermediate list

### ISSUE THREAD

**Issue by jsvd** [elastic/logstash#14500]
The current list of ciphers we set to improve OpenSSL defaults is quite outdated, allowing multiple ciphers that are nowadays deemed weak. It is also preventing new known families of strong ciphers from being used such as CHACHA20 (e.g. TLS_CHACHA20_POLY1305_SHA256).

We should update the list in https://github.com/elastic/logstash/blob/main/logstash-core/lib/logstash/patches/stronger_openssl_defaults.rb#L40 to [Mozilla's intermediate cipher list](https://wiki.mozilla.org/Security/Server_Side_TLS#Intermediate_compatibility_.28recommended.29), currently:

```
TLS_AES_128_GCM_SHA256         TLSv1.3
TLS_AES_256_GCM_SHA384         TLSv1.3
TLS_CHACHA20_POLY1305_SHA256   TLSv1.3
ECDHE-ECDSA-AES128-GCM-SHA256  TLSv1.2
ECDHE-RSA-AES128-GCM-SHA256    TLSv1.2
ECDHE-ECDSA-AES256-GCM-SHA384  TLSv1.2
ECDHE-RSA-AES256-GCM-SHA384    TLSv1.2
ECDHE-ECDSA-CHACHA20-POLY1305  TLSv1.2
ECDHE-RSA-CHACHA20-POLY1305    TLSv1.2
DHE-RSA-AES128-GCM-SHA256      TLSv1.2
DHE-RSA-AES256-GCM-SHA384      TLSv1.2
```

Setting the list above will effectively limit TLS negotiation to TLSv1.2 and TLSv1.3.

This change only affects plugins using JRuby's OpenSSL layer like the TCP output, but doesn't affect plugins using Java's SSL directly (e.g. beats input or tcp input).

**Comment by roaksoax** [comment]
@jsvd seems that the PR has landed, can we close this issue?

**Comment by jsvd** [comment]
@roaksoax no, this is a follow up to the PR. The PR removes 1 particular cipher we know we don't want, but the list is still outdated.
Updating it will be a sort of a breaking change: it may stop existing deployments from working, but on the other hand we should push users to run on ciphers deemed safe to use.

So this change will have to be made with consideration to users whose deployments may stop, by providing guidance on how to re-enable certain certs of ciphers for those that need them (e.g. connecting to endpoints they don't control that use older ciphers).
